packages/nimble-components/src/rich-text-viewer/index.ts

```diff
@@ -11,7 +18,100 @@ declare global {
 /**
  * A nimble styled rich text viewer
  */
-export class RichTextViewer extends FoundationElement {}
+export class RichTextViewer extends FoundationElement {
+    /**
+     * Whether to grow the height of the component to fit the content.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: fit-to-content
+     */
+    @attr({ attribute: 'fit-to-content', mode: 'boolean' })
+    public fitToContent = false;
+
+    /**
+     * @public
+     * @remarks
+     * Accessor: uses the parsing logic to convert the input markdown string to render in the component.
+     */
+    public set markdownValue(value: string) {
+        this._markdownValue = value;
+        this.serializedContent = this.parseMarkdownToDOM(value);
+        this.appendSerializedContentToViewer();
+    }
+
+    /**
+     * @public
+     * @remarks
+     * Accessor: gets the raw markdown string.
+     */
+    public get markdownValue(): string {
+        return this._markdownValue;
+    }
+
+    private _markdownValue = '';
+    private serializedContent?: HTMLElement | DocumentFragment;
+    private readonly markdownParser: MarkdownParser;
+    private readonly domSerializer: DOMSerializer;
+
+    public constructor() {
+        super();
+        this.domSerializer = DOMSerializer.fromSchema(schema);
+        this.markdownParser = this.initializeMarkdownParser();
+    }
+
+    /**
+     * @internal
+     */
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.appendSerializedContentToViewer();
+    }
+
+    private initializeMarkdownParser(): MarkdownParser {
+        /**
+         * It configures the tokenizer of the default Markdown parser with the 'zero' preset.
+         * The 'zero' preset is a configuration with no rules enabled by default to selectively enable specific rules.
+         * https://github.com/markdown-it/markdown-it/blob/master/lib/presets/zero.js#L1
+         *
+         */
+        const zeroTokenizerConfiguration = defaultMarkdownParser.tokenizer.configure('zero');
+
+        // The detailed information of the supported rules were provided in the below CommonMark spec document.
+        // https://spec.commonmark.org/0.30/
+        const supportedTokenizerRules = zeroTokenizerConfiguration.enable([
+            'emphasis',
+            'list',
+            'autolink'
+        ]);
```

Comment:
While playing around in storybook build, couldn't add new line breaks which might be a very obvious need for our commenting use case. 

Do we need to consider adding [line break](https://spec.commonmark.org/0.30/#hard-line-breaks) rule?

Reply:
Need not block this PR necessarily. Can come in a separate PR.